AudioDeviceExample/AudioDevices/ExampleAVAudioEngineDevice.m

```diff
@@ -822,16 +852,10 @@ - (void)handleValidRouteChange {
         _renderingFormat = nil;
         _capturingFormat = nil;
 
-        // Restart the AVAudioEngine with new format
-        TVIAudioFormat *activeFormat = [[self class] activeFormat];
-        if (![activeFormat isEqual:_renderingFormat]) {
-            [self teardownAudioEngine];
-            [self setupAudioEngine];
-        }
```

Comment:
where did this code move? We have to teardown and setup the audio engine with the new formats otherwise audio wont work after format changes. The `setupAudioEngine` takes some time to setup the AVAudioEngine and we have to stop core audio as soon as possible first. So, I was thinking about a couple of solutions - 
1. Call `[self teardownAudioEngine]` and `[self setupAudioEngine];` after `TVIAudioDeviceFormatChanged(context);` is called.
2. Another option is we can teardown the audio unit as soon as possible when `handleValidRouteChange` is called.

Reply:
For now, I'm going to restore this code. I thought it was a mistake but I was indeed the one confused about the AVAudioEngine lifecycle in this class.